modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/rest-openapi.yaml

```diff
@@ -3450,5 +3539,66 @@ paths:
                         Unexpected error
             summary: Creates a WishList Item.
             tags: ["WishListItem"]
+    /attachments/{id}/document:
```

Comment:
for delivery I think it's better to add this to the attachment entity, so that the end user can fetch the contente with one call. Also I believe in this case the URL returned by the API should be blanked because if it doesn't work it doesn't make sense to send it

Reply:
Hi @andrea-ale-sbarra , could you clarify what you meant by URL returned by API being "blanked"?

Also, since the call is meant to be used a nested call, wouldn't it make sense to keep it with Document entity, so we can utilize the following API call format?

`/o/headless-commerce-delivery-catalog/v1.0/channels/channelId/products/productId/attachments
?nestedFields=document`

Sincerely,
Brian I. Kim

Reply:
Hi @brianikim sure:

1. If the attachment is of type "external video", it doesn't make sense to send an URL that point to nothing, indeed part of this bug reports that using that URL falls into an issue. So the idea is to not return a wrong URL (blank it) so that nobody can use it.
2. Here I believe there was a misunderstanding, the nested API call was meant to be a second call from the client to the portal delivery api, but to do that call you need to provide the fileEntryId, that I don't see in this PR. That being said I think that the information about the Document should be carry out with the same call to the Attachment, we don't need to make it a nested field.